### ISSUE TITLE
Feature request: app-level changelogs

### ISSUE THREAD

**Issue by sondrelg** [axodotdev/cargo-dist#396]
Hi again :slightly_smiling_face: 

Is my understanding correct, that in virtual workspaces, only root level changelogs are supported for generating release bodies? 

Would you accept a PR to add support for member-level changelogs? And do you predict it would be hard to add?

**Comment by Gankra** [comment]
As noted in https://github.com/axodotdev/cargo-dist/issues/394#issuecomment-1706854252 I am super down to turn this on, just need you to detail the kind of directory/file structure you think should work.

**Comment by sondrelg** [comment]
I think it might make sense to say "Any folder with a Cargo.toml could have its own CHANGELOG.md" (or whatever name for the file you want to use), and to prefer a workspace member's changelog over the root-level changelog, when it exists.

Not too familiar with how you keep track of workspace members in a virtual workspace, but perhaps there's already some tooling for discovering and/or reading from workspace member directories? I think you already read the Cargo.toml for some configuration values, right?

## Example 1
I would say it makes sense to prefer the more specific changelog, if found
```
|__ apps 
|    |
|    |__ cli app
|           |__ CHANGELOG.md  <-- prefer this
|           |__ Cargo.toml
|
|__ CHANGELOG.md <-- even though this exists
|__ Cargo.toml
```

## Example 2
And otherwise use the less-specific one
```
|__ apps 
|    |
|    |__ cli app  <-- doesn't contain a changelog
|           |__ Cargo.toml
|
|__ CHANGELOG.md  <-- so prefer this
|__ Cargo.toml
```

----

My use-case is a monorepo of web apps, where just a single member is a CLI which I use cargo-dist to publish releases for :+1:

**Comment by Gankra** [comment]
fixed by #393
